weed/server/master_server.go

```diff
@@ -384,17 +402,19 @@ func (ms *MasterServer) OnPeerUpdate(update *master_pb.ClusterNodeUpdate, startF
 					})
 					if err != nil {
 						glog.Warningf("failed to removing old raft server %s: %v", peerName, err)
+						return
 					}
 					glog.V(0).Infof("old raft server %s removed", peerName)
 					return
 				case peerDone := <-ms.onPeerUpdateDoneCn:
 					if peerName == peerDone {
-						glog.V(0).Infof("raft server %s remove canceled", peerName)
+						glog.V(0).Infof("raft server %s remove canceled on onPeerUpdate", peerName)
 						return
 					}
 				}
 			}
 		}(peerName)
+		glog.V(0).Infof("wait %v for raft server %s activity, otherwise delete", RaftServerRemovalTime, peerName)
 		ms.onPeerUpdateDoneCnExist = true
```

Comment:
seems `ms.onPeerUpdateDoneCnExist = true` should be at the beginning of the goroutine.